crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -424,12 +435,37 @@ async fn execute_transactions(
                 periods += 1;
             }
             Ok(Err(err)) => return Err(err),
-            Ok(Ok(_)) => {
+            Ok(Ok(effects)) => {
                 authority_store.insert_executed_transactions(
                     &all_tx_digests,
                     epoch_store.epoch(),
-                    checkpoint_sequence,
+                    checkpoint.sequence_number(),
                 )?;
+
+                let checkpoint_seq_num = checkpoint.sequence_number();
+                let epoch = epoch_store.epoch();
+                let effects: Vec<TransactionEffects> =
+                    effects.into_iter().map(|fx| fx.data().clone()).collect();
+
+                let state = State {
+                    effects,
+                    checkpoint_seq_num,
+                    epoch,
+                    end_of_epoch_flag: checkpoint.next_epoch_committee().is_some(),
+                };
+                accumulator.enqueue(state).await?;
+                // TODO: for now, we block here, as this ensures
+                // that we don't need to handle crash recovery (in
+                // such a case, the checkpoint would get re-executed
+                // and re-enqueued, which is fine). However, for a
+                // checkpoint with 2,000 objects, checkpoint accumulation
+                // could take ~100ms. Therefore we should consider not blocking
```

Comment:
is this really true? that seems surprisingly slow

Reply:
Yeah it would seem so from my testing. I was surprised as well. See `crates/sui-types/src/accumulator.rs::test_benchmark()`

Luckily, creating the root state hash is very fast.

Reply:
Were you running the benchmark in debug mode? I get 7ms (mean) for 1k objects.

Also I think this may not be the best way to write this benchmark - let me push a commit to use a benchmarking tool for this.